security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/request/password/ReactiveOpenIdPasswordAuthenticationProvider.java

```diff
@@ -79,19 +80,19 @@ public ReactiveOpenIdPasswordAuthenticationProvider(OauthClientConfiguration cli
     @Override
     public Publisher<AuthenticationResponse> authenticate(T requestContext, AuthenticationRequest<I, S> authenticationRequest) {
         OpenIdPasswordTokenRequestContext openIdPasswordTokenRequestContext = new OpenIdPasswordTokenRequestContext(authenticationRequest, secureEndpoint, clientConfiguration);
-        return Mono.from(tokenEndpointClient.sendRequest(openIdPasswordTokenRequestContext))
```

Comment:
When using `Publisher` as the return value and needing to chain `flatMap` operators together, I find it more natural to use `Flux` because its `flatMap` operator takes a `Publisher`, whereas the `flatMap` operator of `Mono` requires another `Mono`, thus an additional call to `Mono.from`, introducing another layer of wrapping that makes the call stack deeper.

If we weren't returning `Publisher`, then I would indeed prefer `Mono` in this case.